odoo/tools/mail.py

```diff
@@ -190,6 +190,11 @@ def html_sanitize(src, silent=True, sanitize_tags=True, sanitize_attributes=Fals
     src = src.replace(u'<%', misc.html_escape(u'<%'))
     src = src.replace(u'%>', misc.html_escape(u'%>'))
 
+    # On the specific case of Outlook desktop it adds unnecessary '<o:p></o:p>' tags which are parsed in '<p></p>'
+    # and shows additional lines break when displayed.
+    src = src.replace(u'<o:p>', u'')
+    src = src.replace(u'</o:p>', u'')
```

Comment:
Seems way too specific to put in the generic sanitizer. At least it should be a separate method but I don't think it should really be handled here at all.

<o:p> should not be converted into anything. Putting that tag in my browser directly just creates a 0-width element which shouldn't modify the appearance, afaict

I'll try replicating but seems odd. We'll also really want a test with this, whatever the solution ends up being.

Reply:
For the unclosed <div> tags it's just because I crop that part that I though was helpful to understand the problem.

The <o:p> tag is specific to Microsoft Word to keep a certain format when transferring from word to HTML. 

When you say ", can you identify where that is done and maybe prevent this there instead?" you mean where the <o:p> is converted ?

Reply:
> cropped

Got it

> you mean where <o:p> is converted?

Yes, as far as I can tell if we just insert that into a page it should just remain unrendered, not transform into <p>
So if there's some point where we convert it, would probably be better to just not do that since the reason it's prefixed with with `o:` namespace is because it should only be rendered in outlook/word

Reply:
this happen in the cleaner.clean_html that calls lxml.html.fromstring, for example:

```py
> lxml.html.tostring(lxml.html.fromstring('<o:p>test</o:p>'))
b'<p>test</p>'
> lxml.html.tostring(lxml.html.fromstring('<o:p>test</o:p>', parser=lxml.html.HTMLParser(recover=False)))
*** lxml.etree.XMLSyntaxError: Tag o:p invalid, line 1, column 5
```
if we remove the [recover](https://lxml.de/4.1/api/lxml.etree.HTMLParser-class.html) option (`recover - try hard to parse through broken HTML (default: True)`) the code fails but I don't see an option to keep the "o:p" =/

Reply:
Thanks for having a look!
I see... if we don't use it, it crashes. If we use it, it breaks stuf... lxml is great

I could not find a way to get a clean parsed version of the document with lxml without first fixing the missing namespace (which could be numerous) so indeed we'll have to make do with a simpler approach.

I'd still like to do something more generic than just removing this specific problematic tag. I'll try having a look again tomorrow.